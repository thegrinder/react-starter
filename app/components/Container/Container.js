import styled from 'styled-components';

const Container = styled.div`
  width: 100%;
  margin-left: auto;
  margin-right: auto;
  padding: 0 16px;
  max-width: 960px;
`;

export default Container;
