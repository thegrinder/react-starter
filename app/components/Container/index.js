import styled from 'styled-components';
import { rem } from 'basic-styled-uikit';

const Container = styled.div`
  width: 100%;
  margin-left: auto;
  margin-right: auto;
  padding: 0 ${rem(16)};
  max-width: ${rem(960)};
`;

export default Container;
