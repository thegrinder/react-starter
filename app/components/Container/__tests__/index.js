import React from 'react';
import { ThemeProvider } from 'styled-components';
import { render, cleanup } from 'react-testing-library';
import 'jest-dom/extend-expect';

import theme from '../../../theme/components';
import Container from '..';


const children = <span>children</span>;

const renderComponent = (props = {}) => render(
  <ThemeProvider theme={theme}>
    <Container {...props}>{children}</Container>
  </ThemeProvider>,
);

describe('<Container />', () => {
  afterEach(cleanup);

  it('should render correctly with default props and children', () => {
    const { container: { firstChild }, getByText } = renderComponent();
    const childrenElement = getByText('children');
    expect(firstChild).toBeDefined();
    expect(firstChild).toContainElement(childrenElement);
    expect(firstChild).toMatchSnapshot();
  });

  it('should render <div> tag by default', () => {
    const { container: { firstChild } } = renderComponent();
    expect(firstChild.tagName).toEqual('DIV');
  });
});
