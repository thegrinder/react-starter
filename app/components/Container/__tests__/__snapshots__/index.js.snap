// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should render correctly with default props and children 1`] = `
<div
  class="sc-bdVaJa hGfkPm"
>
  <span>
    children
  </span>
</div>
`;
